Sources/OTel/Logging/Processing/Batch/OTelBatchLogRecordProcessorConfiguration.swift

```diff
@@ -0,0 +1,82 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift OTel open source project
+//
+// Copyright (c) 2024 the Swift OTel project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+/// The configuration options for an ``OTelBatchLogRecordProcessor``.
+@_spi(Logging)
+public struct OTelBatchLogRecordProcessorConfiguration: Sendable {
+    /// The maximum queue size.
+    ///
+    /// - Warning: After this size is reached log will be dropped.
+    public var maximumQueueSize: UInt
+
+    /// The maximum delay between two consecutive log exports.
+    public var scheduleDelay: Duration
+
+    /// The maximum batch size of each export.
+    ///
+    /// - Note: If the queue reaches this size, a batch will be exported even if ``scheduleDelay`` has not elapsed.
+    public var maximumExportBatchSize: UInt
+
+    /// The duration a single export can run until it is cancelled.
+    public var exportTimeout: Duration
+
+    /// Create a batch log processor configuration.
+    ///
+    /// - Parameters:
+    ///   - environment: The environment variables.
+    ///   - maximumQueueSize: A maximum queue size used even if `OTEL_BLRP_MAX_QUEUE_SIZE` is set. Defaults to `2048` if both are `nil`.
+    ///   - scheduleDelay: A schedule delay used even if `OTEL_BLRP_SCHEDULE_DELAY` is set. Defaults to `5` seconds if both are `nil`.
+    ///   - maximumExportBatchSize: A maximum export batch size used even if `OTEL_BLRP_MAX_EXPORT_BATCH_SIZE` is set. Defaults to `512` if both are `nil`.
+    ///   - exportTimeout: An export timeout used even if `OTEL_BLRP_EXPORT_TIMEOUT` is set. Defaults to `30` seconds if both are `nil`.
+    public init(
+        environment: OTelEnvironment,
+        maximumQueueSize: UInt? = nil,
+        scheduleDelay: Duration? = nil,
+        maximumExportBatchSize: UInt? = nil,
+        exportTimeout: Duration? = nil
+    ) {
+        self.maximumQueueSize = environment.requiredValue(
+            programmaticOverride: maximumQueueSize,
+            key: "OTEL_BLRP_MAX_QUEUE_SIZE",
+            defaultValue: 2048,
+            transformValue: UInt.init
+        )
+
+        self.scheduleDelay = environment.requiredValue(
+            programmaticOverride: scheduleDelay,
+            key: "OTEL_BLRP_SCHEDULE_DELAY",
+            defaultValue: .seconds(5),
```

Comment:
It looks like the standardized default schedule delay is different from the one used for span processing. I also checked the other defaults but those are the same as for batch span processing.

```suggestion
            defaultValue: .seconds(1),
```